packages/application/src/services.tsx

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2018-2020 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Dialog, showDialog } from '@jupyterlab/apputils';
+import * as React from 'react';
+
+import { RequestHandler } from './requests';
+
+/**
+ * A utilities class for various elyra services.
+ */
+export class FrontendServices {
+  static noMetadataError(metadataName: string): Promise<Dialog.IResult<any>> {
+    return showDialog({
+      title: 'Error retrieving metadata',
+      body: <p>No {metadataName} metadata has been configured.</p>,
+      buttons: [Dialog.okButton()]
+    });
+  }
+
+  static async getMetadata(namespace: string): Promise<any> {
+    const metadataResponse: any = await RequestHandler.makeGetRequest(
+      'api/metadata/' + namespace,
+      false
+    );
+
+    return metadataResponse[namespace];
```

Comment:
Is the dict format returned here the best option to return, or should we use a list?
Understanding the usage pattern by the UI components would help on this decision (e.g if all consumers loop trough and populate UI components, list would probably make more sense... if there are scenarios where lookups are used, then dict might actually be the best option. Another approach is to enable something like getMetadata().asDict and getMetadata.asList()

Reply:
This currently just returns the metadata "as is" and leaves the formatting to the service functions that call it. Each service function handles error cases and formatting to match how that metadata namespace is used by the frontend.

Reply:
Yes, the "as is" is a dict. And then all the consumers have to convert that to list, at least on couple places I saw.

Reply:
If this call is just calling `GET /api/metadata/{namespace}` we return a JSON object of the form: [`{"runtimes": [ array of metadata instances ]}`](https://github.com/elyra-ai/elyra/blob/master/elyra/api/elyra.yaml#L126-L134), so it's both (a dict and a list).